### PR TITLE
Added Java content 

### LINKED ISSUE #625: [Feature Request] ADD Details for Java Language
## 🚀 Feature Request

there is no content on java language . no intriuction ,  variables , or operators etc. for java

### Describe the solution you'd like
I would like to add details on the topics: Introduction
Variables
Operators
Functions
Loops
Arrays
Objects
Classes
Inheritance
Interfaces
Packages
Multithreading

### Describe alternatives you've considered
Describe each topic theoretically and provide examples with code or pseudocode

---

**Would you like to work on this feature?**  
- [X] Yes  
- [ ] No  

@ajay-dhangar kindly assign me this issue

### PR DESCRIPTION
**Pull request by purnima2904**
### Description
Added detailed content on java including introduction, variables, operators and multithreading with code examples.

Fixes #625 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/languages/java/ex1.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+### 9. **Similarity to C++ and C#**  
+Since Java is syntactically similar to C++ and C#, it is easier for developers to transition between these languages,
+ leveraging existing knowledge to learn Java or vice versa.
+
+
+## Java Syntax
+In this section, we will explore the basic structure of a Java program, using a "Hello World" example to understand key syntax rules.
+
+### 1. "Hello World" Example
+```java
```
Comment (ajay-dhangar) [inline docs/languages/java/ex1.md]: you can try 

````md
```java title="Main.java"
public class Main {
  public static void main(String[] args) {
    System.out.println("Hello World");
  }
}
```
````

File: docs/languages/java/ex1.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+**System.out.println()**
+- Inside the main() method, we can use the println() method to print a line of text to the screen:
+```java
+public static void main(String[] args) {
+  System.out.println("Hello World");
+}
+```
+
+### Note :
+- The curly braces {} marks the beginning and the end of a block of code.
```
Comment (ajay-dhangar) [inline docs/languages/java/ex1.md]: replace `{}` with ``` `{}` ```

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by purnima2904** [comment]
@ajay-dhangar kindly review the changes made as per your request

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**